### PR TITLE
Update dependencies and fix our processor

### PR DESCRIPTION
**Pull request by yaraki**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -128,7 +128,7 @@ subprojects {
             // Treat all Kotlin warnings as errors
             allWarningsAsErrors = true
             // Set JVM target to 1.8
```
Comment (wosys) [inline build.gradle:130]: Maybe this need change as well.

Comment (yrezgui) [inline build.gradle:130]:
```suggestion
            // Set JVM target to 17
```

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -34,18 +34,18 @@ metalavaGradle = "me.tylerbwong.gradle:metalava-gradle:0.1.9"
...
 
-androidx-core = "androidx.core:core-ktx:1.10.1"
-androidx-annotation = "androidx.annotation:annotation:1.6.0"
-androidx-fragment = "androidx.fragment:fragment-ktx:1.6.0"
-androidx-activity = "androidx.activity:activity:1.7.2"
-androidx-activity-compose = "androidx.activity:activity-compose:1.7.2"
+androidx-core = "androidx.core:core-ktx:1.12.0"
+androidx-annotation = "androidx.annotation:annotation:1.7.0"
+androidx-fragment = "androidx.fragment:fragment-ktx:1.6.1"
+androidx-activity = "androidx.activity:activity:1.8.0"
```
Comment (yrezgui) [inline gradle/libs.versions.toml:40]:
```suggestion
androidx-activity = "androidx.activity:activity:1.8.1"
```

Yesterday's fix related to `OnBackPressedDispatcher`

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -34,18 +34,18 @@ metalavaGradle = "me.tylerbwong.gradle:metalava-gradle:0.1.9"
...
+androidx-activity-compose = "androidx.activity:activity-compose:1.8.0"
```
Comment (yrezgui) [inline gradle/libs.versions.toml:41]:
```suggestion
androidx-activity-compose = "androidx.activity:activity-compose:1.8.1"
```

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -15,16 +15,16 @@
 #
 
 [versions]
-compose-bom = "2023.06.00"
-composeCompiler = "1.4.7"
+compose-bom = "2023.10.00"
```
Comment (yrezgui) [inline gradle/libs.versions.toml:18]:
```suggestion
compose-bom = "2023.10.01"
```

**Comment by yaraki** [comment]
I resolved all the comments.

**Review by yrezgui — APPROVED** [review]

**Pull request merged**